client.go

```diff
@@ -33,7 +33,10 @@ func NewClient(ifi *net.Interface) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	return NewClientPacketConn(ifi, p)
+}
 
+func NewClientPacketConn(ifi *net.Interface, p net.PacketConn) (*Client, error) {
```

Comment:
I was sending this up for feedback on where to extend the API, and I wasn't keen to document a method I was unsure of. However, since this is the only feedback, I'm going to assume the content is acceptable, and I've pushed docs.